tests/test_interconnect/test_interconnect_cgra.py

```diff
@@ -97,57 +75,32 @@ def test_interconnect_point_wise(batch_size: int):
-    mem_out = graph16.get_port(2, 1, "data_out")
-    config_data.append(interconnect.get_node_bitstream_config(input_node,
-                                                              mem_in))
-    next_node0 = graph16.get_sb(2, 1, SwitchBoxSide.WEST, 0, SwitchBoxIO.SB_OUT)
-    config_data.append(interconnect.get_node_bitstream_config(input_node,
-                                                              next_node0))
-    # route the line buffer output to the track 1
-    next_node1 = graph16.get_sb(2, 1, SwitchBoxSide.WEST, 1, SwitchBoxIO.SB_OUT)
-    config_data.append(interconnect.get_node_bitstream_config(mem_out,
-                                                              next_node1))
-    # route the result to the input
-    next_node0 = graph16.get_sb(1, 1, SwitchBoxSide.EAST, 0, SwitchBoxIO.SB_IN)
-    # to data 0
-    data0 = graph16.get_port(1, 1, "data0")
-    config_data.append(interconnect.get_node_bitstream_config(next_node0,
-                                                              data0))
-    next_node1 = graph16.get_sb(1, 1, SwitchBoxSide.EAST, 1, SwitchBoxIO.SB_IN)
-    data1 = graph16.get_port(1, 1, "data1")
-    config_data.append(interconnect.get_node_bitstream_config(next_node1,
-                                                              data1))
-
-    # route the pe out to the (1, 1) left sb T0
-    data_out = graph16.get_port(1, 1, "res")
-    output_node = graph16.get_sb(1, 1, SwitchBoxSide.WEST, 0,
-                                 SwitchBoxIO.SB_OUT)
-    config_data.append(interconnect.get_node_bitstream_config(data_out,
-                                                              output_node))
-
-    # in this case we configure (2, 1) as line buffer mode
-    config_data.append((0x00000201, 0x00000004 | (depth << 3)))
-    # then (1, 1) is configured as add
-    config_data.append((0xFF000101, 0x000AF000))
-
-    # also need to wire a constant 1 to wen
-    wen_sb = graph1.get_sb(2, 1, SwitchBoxSide.NORTH, 1, SwitchBoxIO.SB_IN)
-    wen_port = graph1.get_port(2, 1, "wen_in")
-    config_data.append(interconnect.get_node_bitstream_config(wen_sb,
-                                                              wen_port))
+
+    netlist = {
+        "e0": [("I0", "io2f_16"), ("m0", "data_in"), ("p0", "data0")],
+        "e1": [("m0", "data_out"), ("p0", "data1")],
+        "e3": [("p0", "res"), ("I1", "f2io_16")],
+        "e4": [("i0", "io2f_1"), ("m0", "wen_in")]
+    }
+    bus = {"e0": 16, "e1": 16, "e3": 16, "e4": 1}
+
+    placement, routing = pnr(interconnect, (netlist, bus))
+    config_data = interconnect.get_route_bitstream(routing)
+
+    # in this case we configure m0 as line buffer mode
+    mem_x, mem_y = placement["m0"]
+    config_data.append((0x00000000 | (mem_x << 8 | mem_y),
+                        0x00000004 | (depth << 3)))
+    # then p0 is configured as add
+    pe_x, pe_y = placement["p0"]
+    config_data.append((0xFF000000 | (pe_x << 8 | pe_y), 0x000AF000))
```

Comment:
I don't think this is related to PnR/archipelago, but at some point it would be good to abstract the logic for generating the configuration data (e.g. a method interface, so, for example, if we decide to change the bit positions/layout of the various fields, this test will still work.

Reply:
Agreed. I will refactor this later as a separate PR. There are lots of places in the code base that assume this bit layout.